src/baldr/nodeinfo.cc

```diff
@@ -221,10 +221,15 @@ void NodeInfo::set_stop_index(const uint32_t stop_index) {
 // Get the heading of the local edge given its local index. Supports
 // up to 8 local edges. Headings are expanded from 8 bits.
 uint32_t NodeInfo::heading(const uint32_t localidx) const {
-  // Make sure everything is 64 bit!
-  uint64_t shift = localidx * 8; // 8 bits per index
-  return static_cast<uint32_t>(std::round(
-      ((headings_ & (static_cast<uint64_t>(255) << shift)) >> shift) * kHeadingExpandFactor));
+  if (localidx > kMaxLocalEdgeIndex) {
+    LOG_WARN("Local index exceeds max in set_heading, return 0");
+    return 0;
+  } else {
+    // Make sure everything is 64 bit!
+    uint64_t shift = localidx * 8; // 8 bits per index
+    return static_cast<uint32_t>(std::round(
+        ((headings_ & (static_cast<uint64_t>(255) << shift)) >> shift) * kHeadingExpandFactor));
```

Comment:
Also a problem with shift exponent. At `set` function too big `localidx` is skipped (see [the line](https://github.com/SvetlanaBayarovich/valhalla/pull/1/files#diff-ee1ff2c003c7e0c299317cf7b4760e3cL234)).
I added the same check but I'm not 100% sure what the return should be.

Reply:
related.  https://github.com/valhalla/valhalla/issues/2494

Why would we be calling this with a high index?

Reply:
can only store headings for 8 local indexes and the if statement protects if localindex > 7. The node based headings probably need to be deprecated. These headings are only used in map matching (could use simplified turn costing logic) and during data creation (can create temporary sequence files to look up headings to all edges).